### ISSUE TITLE
Add pipe2 detection and substitute pipe2 function

### ISSUE THREAD

**Issue by mdblack98** [Hamlib/Hamlib#904]
We can detect the pipe2 function being available for iofunc.c async operations.
If not we can substitute our own pipe2 function.
Would clean up the code a bit during async opening.

**Comment by mikaelnousiainen** [comment]
@mdblack98 Is this still relevant after #918 Windows implementation?

**Comment by mdblack98** [comment]
Yes -- would make for some code reduction since it's duplicated.
Easy change just to implement our own pipe2 so it's portable.
